src/Kind/Parse.hs

```diff
@@ -961,8 +970,11 @@ flattenDef rules depth =
 flattenWith :: Int -> With -> Term
 flattenWith dep (WBod bod)     = bod
 flattenWith dep (WWit wth rul) =
-  let bod = flattenDef rul (dep + 1)
-  in foldl App bod wth
+  -- Wrap the 'with' arguments and patterns in Pairs since the type checker only takes one match argument.
+  let wthA = foldr1 (\x acc -> Con "Pair" [(Nothing, x), (Nothing, acc)]) wth
```

Comment:
Is the `Pair` constructor always available to the type checker? Like, does it need to be imported?

Reply:
It needs to be in scope, just like String and List. Kind and kindbook are expected to always be in sync